src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java

```diff
@@ -909,15 +909,19 @@ public DataFetcher<Iterable<Route>> routes() {
       if (args.getLegacyGraphQLName() != null) {
         String name = args.getLegacyGraphQLName().toLowerCase(environment.getLocale());
         routeStream =
-          routeStream.filter(route ->
-            Stream
-              .of(route.getShortName(), route.getLongName())
-              .filter(Objects::nonNull)
-              .map(s -> s.toLowerCase(environment.getLocale()))
-              .anyMatch(s -> s.startsWith(name))
-          );
+          routeStream.filter(route -> {
+            if (
+              route.getShortName() != null &&
+              LegacyGraphQLUtils.startsWith(route.getShortName(), name, environment.getLocale())
+            ) {
+              return true;
+            }
+            return (
+              route.getLongName() != null &&
+              LegacyGraphQLUtils.startsWith(route.getLongName(), name, environment.getLocale())
+            );
+          });
```

Comment:
If we add a null check in `LegacyGraphQLUtils.startsWith`, this could be simplified down to

```suggestion
          routeStream.filter(route -> 
            LegacyGraphQLUtils.startsWith(route.getShortName(), name, environment.getLocale()) ||
            LegacyGraphQLUtils.startsWith(route.getLongName(), name, environment.getLocale())
          );
```

Reply:
Done